src/thor/expansion_action.cc

```diff
@@ -104,50 +110,51 @@ std::string thor_worker_t::expansion(Api& request) {
   // a lambda that the path algorithm can call to add stuff to the dom
   // route and isochrone produce different GeoJSON properties
   std::string algo = "";
-  auto track_expansion = [&](baldr::GraphReader& reader, baldr::GraphId edgeid,
-                             baldr::GraphId prev_edgeid, const char* algorithm = nullptr,
-                             const Expansion_EdgeStatus status = Expansion_EdgeStatus_reached,
-                             const float duration = 0.f, const uint32_t distance = 0,
-                             const float cost = 0.f) {
-    algo = algorithm;
-
-    auto tile = reader.GetGraphTile(edgeid);
-    if (tile == nullptr) {
-      LOG_ERROR("thor_worker_t::expansion error, tile no longer available" +
-                std::to_string(edgeid.Tile_Base()));
-      return;
-    }
-
-    // if requested, skip this edge in case its opposite edge has been added
-    // before (i.e. lower cost) else add this edge's id to the lookup container
-    if (skip_opps) {
-      auto opp_tile = tile;
-      auto opp_edgeid = reader.GetOpposingEdgeId(edgeid, opp_tile);
-      if (opp_edgeid && opp_edges.count(opp_edgeid))
-        return;
-      opp_edges.insert(edgeid);
-    }
+  auto track_expansion =
+      [&](baldr::GraphReader& reader, baldr::GraphId edgeid, baldr::GraphId prev_edgeid,
+          const char* algorithm = nullptr,
+          const Expansion_EdgeStatus status = Expansion_EdgeStatus_reached,
+          const float duration = 0.f, const uint32_t distance = 0, const float cost = 0.f,
+          const Expansion_ExpansionType expansion_type = Expansion_ExpansionType_forward) {
+        algo = algorithm;
```

Comment:
aw man, these diffs again... all that changed here is the extra `expansion_type` argument that's being passed around, and it has some effects on the formatting.